pkgs/stdenv/freebsd/default.nix

```diff
@@ -1,276 +1,460 @@
+    x86_64-freebsd = {
+      bash = {
+        url = "https://f005.backblazeb2.com/file/nixbsd/aldkfyqiq0nspljgl5pr0pgvxq99qirk-x86_64-freebsd-bash";
+        hash = "sha256-6keIiXGHkqtaNM1A+ysI3H4FUmbQcjY6RMq+ET9ImQE=";
+      };
+      mkdir = {
+        url = "https://f005.backblazeb2.com/file/nixbsd/f00mn41pls87615px9qjzhxkrcyi2wxc-x86_64-freebsd-mkdir";
+        hash = "sha256-rPNJ8R4EgyXsdsgxUKg05JbtoWrsd+3Txlhn2fiyan4=";
+      };
+      tar = {
+        url = "https://f005.backblazeb2.com/file/nixbsd/0j1x2kvmb1j2fi9jsmlaxxpg22mj2l59-x86_64-freebsd-tar";
+        hash = "sha256-qhc9sfH/IpB7XodEwF+sU7784gyBtZtvL+KUnzjboBA=";
+      };
+      unxz = {
+        url = "https://f005.backblazeb2.com/file/nixbsd/y14fgwj3080h9bqdv5llsfywnq9gj2dx-x86_64-freebsd-unxz";
+        hash = "sha256-it5W1qmVMa6fChgf6m6gWgJSoqxLA3pcx8dus3BDhSE=";
+      };
+      chmod = {
+        url = "https://f005.backblazeb2.com/file/nixbsd/iab27r5x6jaklqg11s4rib7kb4fi1hkf-x86_64-freebsd-chmod";
+        hash = "sha256-iBdzCXKU+03MfOA75ASrXOD3arPtQnvKqsdCf32oKxk=";
+      };
+      bootstrapFiles = {
+        url = "https://f005.backblazeb2.com/file/nixbsd/wvbr4if59gqhc537ai7g98p4s0nxds85-x86_64-freebsd-bootstrap-files.tar.xz";
+        hash = "sha256-vYp8sGHg/Mu5nzQpZI2j9MInDydSQhaJYt8kMuSiLoM=";
+        name = "bootstrap-files.tar.xz";
+        executable = false;
+      };
     };
-    gawk = trivialBuilder rec {
-      inherit (localSystem) system;
-      name = "gawk";
-      ver = "5.0.1";
-      url = "https://ftp.gnu.org/gnu/${name}/${name}-${ver}.tar.xz";
-      sha256 = "15570p7g2x54asvr2fsc56sxzmm08fbk4mzpcs5n92fp9vq8cklf";
-      configureArgs = [ "--disable-nls"
-                        "--disable-mpfr"
-                        "--without-libintl-prefix"
-                        "--without-libiconv-prefix"
-                        "--without-libsegsegv-prefix"
-                      ];
+  };
+  fetchurlBoot = import <nix/fetchurl.nix>;
+  all-bootstrap-urls = all-bootstrap-urls-table.${localSystem.system};
+  all-bootstrap-files = lib.mapAttrs (a: v: fetchurlBoot (v // {
+    executable = v.executable or true;
+    name = v.name or a;
+  })) all-bootstrap-urls;
+  mkExtraBuildCommands0 = cc: ''
+    rsrc="$out/resource-root"
+    mkdir "$rsrc"
+    ln -s "${cc.lib}/lib/clang/16/include" "$rsrc"
+    echo "-resource-dir=$rsrc" >> $out/nix-support/cc-cflags
+  '';
+  mkExtraBuildCommands = cc: compiler-rt: mkExtraBuildCommands0 cc + ''
+    ln -s "${compiler-rt.out}/lib" "$rsrc/lib"
+    ln -s "${compiler-rt.out}/share" "$rsrc/share"
+  '';
+in
+[
+  ({}: let
+    bootstrap-tools = (derivation {
+      inherit system;
+      name = "bootstrap-tools";
+      pname = "bootstrap-tools";
+      version = "9.9.9";
+      builder = all-bootstrap-files.bash;
+      args = [ ./unpack-bootstrap-files.sh ];
+      inherit (all-bootstrap-files) tar unxz mkdir chmod bootstrapFiles;
+    });
+
+    linkBS = (attrs: derivation (attrs // {
```

Comment:
Interesting name choice here ;)

Reply:
it's how i feel about the pure bootstrapping process -_-